CPAC/utils/monitoring/custom_logging.py

```diff
@@ -55,7 +57,75 @@ def getLogger(name):  # pylint: disable=invalid-name
     """
     if name in MOCK_LOGGERS:
         return MOCK_LOGGERS[name]
-    return logging.getLogger(name)
+    logger = nipype_logging.getLogger(name)
+    return logging.getLogger(name) if logger is None else logger
+
+
+def log_failed_subprocess(cpe):
+    """Pass STDERR from a subprocess to the interface's logger
+
+    Parameters
+    ----------
+    cpe : subprocess.CalledProcessError
+    """
+    logger = getLogger('nipype.interface')
+    logger.error("%s\nExit code %s", cpe.output, cpe.returncode)
+
+
+def log_subprocess(cmd, *args, raise_error=True, **kwargs):
+    """Pass STDERR and STDOUT from subprocess to interface's logger.
+    This function is nearly a drop-in replacement for
```

Comment:
Interesting addition - re: nearly, is the only possible issue the caveat below?

Reply:
Yeah. `subprocess.check_output` just returns `stdout`, but I thought it could be useful to get the exit code too.

I could easily be convinced to make it an actual drop-in replacement though. We could do like the original function and raise an exception if the exit code ≠ 0, or we could get fancy and make a subclass of string that has an `exit_code` property

Reply:
I am tempted by these ideas, want to just extend it next release and merge this version in?

Reply:
Yeah, whatever works for you is fine with me. I used it [here](https://github.com/FCP-INDI/C-PAC/blob/78a775877e626e03f0841b5bce4e08d1beb2a61a/CPAC/surface/surf_preproc.py#L43-L66) in this PR looking at https://github.com/FCP-INDI/C-PAC/issues/1870, informing https://github.com/FCP-INDI/C-PAC/pull/1871 with @amygutierrez.

And I don't know if @tergeorge is (planning on) including this kind of logging in any of her current / upcoming PRs, but I know she's been using it to debug some of the code she's been working on.

All that is to say, after merging this (and maybe some of Teresa's code), there will be a small number of calls to this function to update if we decide to change the return(s).